src/libYARP_dev/src/modules/RGBDSensorClient/RGBDSensorClient.h

```diff
@@ -0,0 +1,339 @@
+ * | synchPolicy     |      -         | string  |  -             |  latest       | No                             | Choose the policy to use to synch color and depth frame together               | Values are 'latest', 'seqNum', 'time' (this may require an addictional parameter)|
+ * | watchdog        |      -         | double  |  ms            |   -           | Yes                            | Verify refresh of data on ports whitin this time, otherwise throws an error    |  - |
+ *
+ * Configuration file using .ini format, using subdevice keyword.
+ *
+ * \code{.unparsed}
+ * device RGBDSensorClient
+ * localImagePort     /localImagePort
+ * localDepthPort     /localDepthPort
+ * remoteImagePort    /iCub/colorCamera
+ * remoteDepthPort    /iCub/depthCamera
+ * watchdog 200
+ * \endcode
+ *
+ * XML format, using 'networks' keywork. This file is meant to be used in junction with robotInterface executable,
+ * therefore has an addictional section at the end.
+ *
+ * \code{.xml}
+ *  <!-- Following parameters are meaningful ONLY for robotInterface -->
+ *
+ * <param name="localImagePort">    /<robotName>/localImagePort    </param>
+ * <param name="localDepthPort">    /<robotName>/localDepthPort    </param>
+ * <param name="remoteImagePort">   /<robotName>/colorCamera       </param>
+ * <param name="remoteDepthPort">   /<robotName>/depthCamera       </param>
+ * <param name="watchdog">          200                            </param>
+ * \endcode
+ *
+ */
+
+class yarp::dev::RGBDSensorClient:  public DeviceDriver,
+                                    public IRGBDSensor
+{
+#ifndef DOXYGEN_SHOULD_SKIP_THIS
+protected:
+    yarp::os::ConstString local_colorFrame_StreamingPort_Name;
+    yarp::os::ConstString local_depthFrame_StreamingPort_Name;
+    yarp::os::ConstString remote_colorFrame_StreamingPort_Name;
+    yarp::os::ConstString remote_depthFrame_StreamingPort_Name;
+    yarp::os::BufferedPort<yarp::sig::FlexImage> colorFrame_StreamingPort;
+    yarp::os::BufferedPort<yarp::sig::FlexImage> depthFrame_StreamingPort;
+
+    yarp::os::ConstString local_colorFrame_rpcPort_Name;
+    yarp::os::ConstString local_depthFrame_rpcPort_Name;
+    yarp::os::ConstString remote_colorFrame_rpcPort_Name;
+    yarp::os::ConstString remote_depthFrame_rpcPort_Name;
+    yarp::os::Port colorFrame_rpcPort;
+    yarp::os::Port depthFrame_rpcPort;
+
+    // It should be possible to attach this guy to more than one port, try to see what
+    // will happen when receiving 2 calls at the same time (receive one calls while serving
+    // another one, it will result in concurrent thread most probably) and buffering issues.
+
+    // Image data specs
+    int hDim, vDim;
+    int _rate;
+    std::string sensorId;
+    yarp::dev::IRGBDSensor *sensor_p;
+    IRGBDSensor::RGBDSensor_status _sensorStatus;
```

Comment:
Is there a reason why `_rate` and  `_sensorStatus` start with `_` and the other members don't?
If not, maybe you should be keeping the same naming convention...